cache/contenthash/checksum_test.go

```diff
@@ -554,6 +558,108 @@ func TestChecksumIncludeExclude(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestChecksumIncludeDoubleStar(t *testing.T) {
+	t.Parallel()
+	tmpdir, err := ioutil.TempDir("", "buildkit-state")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpdir)
+
+	snapshotter, err := native.NewSnapshotter(filepath.Join(tmpdir, "snapshots"))
+	require.NoError(t, err)
+	cm, _ := setupCacheManager(t, tmpdir, "native", snapshotter)
+	defer cm.Close()
+
+	ch := []string{
+		"ADD prefix dir",
+		"ADD prefix/a dir",
+		"ADD prefix/a/b dir",
+		"ADD prefix/a/b/c dir",
+	}
+
+	ref := createRef(t, cm, ch)
+
+	cc, err := newCacheContext(ref.Metadata(), nil)
+	require.NoError(t, err)
+
+	dgst, err := cc.Checksum(context.TODO(), ref, "prefix/a", ChecksumOpts{IncludePatterns: []string{"**/foo/**"}}, nil)
+	require.NoError(t, err)
+	// Nothing included
+	require.Equal(t, digest.FromBytes([]byte{}), dgst)
+
+	// Same, with Wildcard = true
+	dgst, err = cc.Checksum(context.TODO(), ref, "prefix/a", ChecksumOpts{IncludePatterns: []string{"**/foo/**"}, Wildcard: true}, nil)
+	require.NoError(t, err)
+	require.Equal(t, digest.FromBytes([]byte{}), dgst)
+
+	ch = []string{
+		"ADD prefix dir",
+		"ADD prefix/a dir",
+		"ADD prefix/a/b dir",
+		"ADD prefix/a/b/c dir",
+		"ADD prefix/a/b/c/foo dir",
+		"ADD prefix/a/b/c/foo/bar file abc",
+	}
+
+	ref = createRef(t, cm, ch)
+
+	cc, err = newCacheContext(ref.Metadata(), nil)
+	require.NoError(t, err)
+
+	dgst, err = cc.Checksum(context.TODO(), ref, "prefix/a", ChecksumOpts{IncludePatterns: []string{"**/foo/**", "**/report"}}, nil)
+	require.NoError(t, err)
+	// Now there is a file included
+	require.NotEqual(t, digest.FromBytes([]byte{}), dgst)
```

Comment:
Would it make sense to add constant expected values to at least some of these to detect regressions?

Reply:
Done.